src/main/kotlin/ch/hevs/cloudio/cloud/security/CloudioPermissionManager.kt

```diff
@@ -111,25 +111,36 @@ class CloudioPermissionManager(
     }
 
     fun resolveEndpointModelElementPermission(userDetails: CloudioUserDetails, modelID: ModelIdentifier): EndpointModelElementPermission {
-        var permission = EndpointModelElementPermission.DENY
+        var allPermissionsList = getAllEndpointModelElementPermissions(userDetails, modelID.endpoint)
 
-        userEndpointPermissionRepository.findByUserIDAndEndpointUUID(userDetails.id, modelID.endpoint).ifPresent {
-            it.modelPermissions[modelID.toModelPath()]?.apply {
-                permission = permission.higher(this)
-            }
+        //add the permissions with the most precise modelPath to the permissionsList
+        var temp = modelID.toString()
```

Comment:
IIRC the `ModelIdentifier` class should have methods to do such tasks.

Reply:
I didn't find such methods